File: app/src/main/java/cc/smartcash/wallet/Services/WalletService.java
package cc.smartcash.wallet.Services;

import cc.smartcash.wallet.Models.SendPayment;
import cc.smartcash.wallet.Models.TransactionResponse;
import cc.smartcash.wallet.Models.WalletPaymentFeeRequest;
import cc.smartcash.wallet.Models.WebWalletRootResponse;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface WalletService {

    @POST("wallet/sendpayment")
    Call<TransactionResponse> sentPayment(@Header("Authorization") String auth, @Body SendPayment sendPayment);

    @POST("wallet/paymentfee")
    Call<WebWalletRootResponse<Double>> getFee(@Header("Authorization") String auth, @Body WalletPaymentFeeRequest feeRequest);
}

File: app/src/main/java/cc/smartcash/wallet/Services/WalletService.kt
package cc.smartcash.wallet.Services

import cc.smartcash.wallet.Models.SendPayment
import cc.smartcash.wallet.Models.WalletPaymentFeeRequest
import cc.smartcash.wallet.Models.WebWalletRootResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.POST

interface WalletService {

    @POST("wallet/sendpayment")
    fun sentPayment(@Header("Authorization") auth: String, @Body sendPayment: SendPayment): Call<WebWalletRootResponse<String>>

    @POST("wallet/paymentfee")
    fun getFee(@Header("Authorization") auth: String, @Body feeRequest: WalletPaymentFeeRequest): Call<WebWalletRootResponse<Double>>
}